### PR TITLE
add setting rke-version

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/15192

### CODE REVIEW

File: scripts/build-server [hunk 1]
```diff
@@ -7,4 +7,13 @@ cd $(dirname $0)/..
...
+
+while IFS=' ' read -r field1 field2
+do if [[ $field1 = *"rke"* ]] 
+	then 
+	RKEVersion=$field2
+	break
+	fi
+done<vendor.conf
+
+CGO_ENABLED=0 go build -i -tags k8s -ldflags "-X main.VERSION=$VERSION -X github.com/rancher/rancher/pkg/settings.RKEVersion=$RKEVersion $LINKFLAGS" -o bin/rancher
```
Comment (alena1108) [inline scripts/build-server]: do we have to specify the absolute path? `main` is defined just like that

Comment (alena1108) [reply scripts/build-server]: nvm, it's a go imports package

File: scripts/build-server [hunk 2]
```diff
@@ -7,4 +7,13 @@ cd $(dirname $0)/..
 
 mkdir -p bin
 [ "$(uname)" != "Darwin" ] && LINKFLAGS="-linkmode external -extldflags -static -s"
-CGO_ENABLED=0 go build -i -tags k8s -ldflags "-X main.VERSION=$VERSION $LINKFLAGS" -o bin/rancher
+
+while IFS=' ' read -r field1 field2
```
Comment (alena1108) [inline scripts/build-server]: can we consider using grep/awk to fetch the needed information from the file?

Comment (ibuildthecloud) [reply scripts/build-server]:
```bash
RKE_VERSION=$(grep '^github.com/rancher/rke' vendor.conf | awk '{print $2}')
```

Please use RKE_VERSION for variable as it matches shell scripting conventions for variables

**Pull request merged**